docs/source/API-Doc.rst

```diff
@@ -0,0 +1,724 @@
+
+       :return: list[Hyperparamter]
+
+   .. method:: get_default_configuration()
+
+       Returns a configuration containing hyperparameters with default values
+
+       :return: Configuration
+
+   .. method:: check_configuration(configuration)
+
+       :param Configuration configuration:
+
+
+   .. method:: check_configuration_vector_representation(vector)
+
+       :param np.ndarray vector:
+
+   .. method:: sample_configuration(size=1)
+
+       sample ``size`` configurations
+
+       :param int size:
+       :return: Configuration,list[Configuration]
+
+   .. method:: seed(seed)
+
+       Sets the random seed.
+
+       :param int seed:
+
+Configuration
+=============
+
+.. autoclass:: ConfigSpace.configuration_space.Configuration
+
+.. _Hyperparameters:
+
+Hyperparameters
+===============
+
+In this section, the different types of hyperparameters are introduced. ConfigSpace is able to handle integer, float, categorical as well as ordinal hyperparameters.
+Float and integer hyperparameters are available as **uniform** or **normal distributed** ones.
+So when a hyperparameter is sampled from the configuration space, its value is distributed according to the specified type.
+
+Example usages are shown in the :doc:`quickstart <quickstart>`
+
+3.1 Integer hyperparameters
+---------------------------
+
+.. py:class:: ConfigSpace.hyperparameters.UniformIntegerHyperparameter(name, lower, upper, default_value=None, q=None, log=False) -> None:
+
+    Creates an integer hyperparameter with values sampled from a uniform distribution with values from ``lower`` to ``upper``
+
+    :param str name: Name of the hyperparameter with which it can be accessed.
+    :param int lower: lower bound of a range of values from which the hyperparameter will be sampled.
+    :param int upper: upper bound.
+    :param int,None default_value: Sets the default value of a hyperparameter to a given value.
+    :param None,int q: Quantization factor
+    :param bool log: If ``True``, the values of the hyperparameter will be sampled on a logarithmic scale.
+
+**Example**::
+
+    import ConfigSpace as CS
+    import ConfigSpace.hyperparameters as CSH
+
+    cs = CS.ConfigurationSpace()
+    uniform_integer_hp = CSH.UniformIntegerHyperparameter('uni_int', lower=10, upper=100, log=False)
+
+    cs.add_hyperparameter(uniform_integer_hp)
+
+.. py:class:: ConfigSpace.hyperparameters.NormalIntegerHyperparameter(name, mu, sigma, default_value=None, q=None, log=False) -> None:
+
+    Creates an integer hyperparameter with values sampled from a normal distribution :math:`\mathcal{N}(\mu, \sigma^2)`
+
+    :param str name: Name of the hyperparameter with which it can be accessed.
+    :param int mu: Mean of the distribution.
+    :param int,float sigma: Standard deviation of the distribution.
+    :param int,None default_value: Sets the default value of a hyperparameter to a given value.
+    :param None,int q: Quantization factor
```

Comment:
Something seems to be wroing with the formatting here and in the two lines above.

Reply:
Hm, i can't really see a wrong behaviour in the html output. 

![image](https://user-images.githubusercontent.com/17492460/48660247-2c403c00-ea5f-11e8-8166-4ab36fa123e6.png)

Reply:
Interesting, then only github doesn't render this correctly.